src/Context/ProductContext.js

```diff
@@ -1,20 +1,37 @@
 import { useContext, createContext, useReducer } from "react";
-import { brands } from "../data"
 import { productReducer } from "../Reducer";
 import { sortData, filterGender, filterInStock, filteredBrands, filterPrice  } from "../utils";
 
+
 const productContext = createContext(null);
 
 const ProductContext = ({children}) => {
-    const [{ products, sortBy, gender, brandFilter, priceFilter, inStock }, dispatch] = useReducer(productReducer, {
+    const brandInitialState = {"dior": false, "davidoff": false, "versace": false, "calvin klien": false, "gucci": false, "nykaa": false, "engage": false}
```

Comment:
suggestion: use codeformatter to format on save of the file automatically and install prettier if not already there in your code editor